app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  def self.update_user_post_count(user)
+    if user.posts_counter.nil?
+      user.posts_counter = 1
+    else
+      user.posts_counter += 1
+    end
+  end
```

Comment:
- [ ] Optional suggestion

I recommend using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_user_post_count` method every time a new post has been saved to the database, then make the `update_user_post_count` method private

```ruby
  after_save :update_user_post_count

  private

  def self.update_user_post_count
    author.increment!(:posts_counter)
  end
```